payloads/libpayload/arch/x86/multiboot.c

```diff
@@ -48,6 +48,8 @@ static int mb_add_memrange(struct sysinfo_t *info, unsigned long long base,
 
 	info->memrange[info->n_memranges].base = base;
 	info->memrange[info->n_memranges].size = size;
+	if (type >= CB_MEM_VENDOR_RSVD)
```

Comment:
payload defines 6 types of memory,  and loader should provide one of these 6 types. And type of memory is passed from EBX by loader, which means loader provides invalid memory type.

Why loader will provide these invalid memory type?

Reply:
per multiboot spec:
where a value of 1 indicates available RAM, value of 3 indicates usable memory holding ACPI information, value of 4 indicates reserved memory which needs to be preserved on hibernation, value of 5 indicates a memory which is occupied by defective RAM modules and **all other values currently indicated a reserved area**.

BTW, grub defines below type:
typedef enum grub_memory_type
  {
    GRUB_MEMORY_AVAILABLE = 1,
    GRUB_MEMORY_RESERVED = 2,
    GRUB_MEMORY_ACPI = 3,
    GRUB_MEMORY_NVS = 4,
    GRUB_MEMORY_BADRAM = 5,
    GRUB_MEMORY_PERSISTENT = 7,
    GRUB_MEMORY_PERSISTENT_LEGACY = 12,
    GRUB_MEMORY_COREBOOT_TABLES = 16,
    GRUB_MEMORY_CODE = 20,
    /* This one is special: it's used internally but is never reported
       by firmware. Don't use -1 as it's used internally for other purposes. */
    GRUB_MEMORY_HOLE = -2,
    GRUB_MEMORY_MAX = 0x10000
  } grub_memory_type_t;

Reply:
LGTM

Reply:
@zhangckid could you please make a minor change:
          if (type >CB_MEM_VENDOR_RSVD)